### PR TITLE
[Serializer] Add support for denormalizing invalid datetime without throwing an exception

### LINKED ISSUE #27824: [Serializer] Do not throw exception in the DateTimeNormalizer if it's not a date
**Description**  
ATM, if the client send a malformated date (or something else) in a field mapped as `DateTime`, an exception occurs.

```
Symfony\Component\Serializer\Exception\NotNormalizableValueException:
DateTimeImmutable::__construct(): Failed to parse time string (2018-08-0111111) at position 14 (1): Unexpected character

  at vendor/symfony/symfony/src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php:116
  at Symfony\Component\Serializer\Normalizer\DateTimeNormalizer->denormalize('2018-08-0111111', 'DateTimeImmutable', 'json', array('cache_key' => '6de9eb2685609b8387658584987a05b0'))
     (vendor/symfony/symfony/src/Symfony/Component/Serializer/Serializer.php:182)
  at Symfony\Component\Serializer\Serializer->denormalize('2018-08-0111111', 'DateTimeImmutable', 'json', array('cache_key' => '6de9eb2685609b8387658584987a05b0'))

# ...
```

By default this Exception is not catched and leads to a 500
Even If we put some validation, the validation comes **after** ... So this is not related

**Example**  
IMHO, it would be better to not throw an exception, then the validator (if exist) will add a violation on the property

---

Maybe this idea could be extended to others normalizer

---

ping @dunglas

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | yes
| Tickets       | Fix #27824
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by Simperfit — APPROVED** [review]

File: src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php [hunk 1]
```diff
@@ -39,6 +43,10 @@ class DateTimeNormalizer implements NormalizerInterface, DenormalizerInterface,
     public function __construct(array $defaultContext = [])
     {
         $this->defaultContext = array_merge($this->defaultContext, $defaultContext);
+
+        if (null === $this->defaultContext[self::THROW_EXCEPTION_ON_INVALID_KEY]) {
+            trigger_deprecation('symfony/serializer', '5.4', 'The key context "%s" of "%s" must be defined. The value will be "false" in Symfony 6.0.', self::THROW_EXCEPTION_ON_INVALID_KEY, __CLASS__);
```
Comment (OskarStark) [inline src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php:48]:
```suggestion
            trigger_deprecation('symfony/serializer', '5.4', 'The key context "%s" of "%s" must be provided. The value will default to "false" in Symfony 6.0.', self::THROW_EXCEPTION_ON_INVALID_KEY, __CLASS__);
```

**Comment by lyrixx** [comment]
**Warning** Don't merge it for now, I think I found a better and more global way to fix this issue

**Comment by fabpot** [comment]
@lyrixx Can we close this one?

**Comment by lyrixx** [comment]
Yes, Sorry for the late reply. I wanted to be sure #42502 is gonna be merged, but there are already 2 👍🏼 So I think it's safe to close this one.

**Pull request closed without merging**